realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -49,7 +49,7 @@
     // Caches Class objects (both model classes and proxy classes) to their Schema object
     private final Map<Class<? extends RealmModel>, RealmObjectSchema> classToSchema = new HashMap<>();
     // Caches Class Strings to their Schema object
-    private final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
+    final Map<String, RealmObjectSchema> dynamicClassToSchema = new HashMap<>();
```

Comment:
If you are going to make this other than private, you need to control access to it.  Please wrap it in an `immutableHashMap`.